### ISSUE TITLE
Problem with the basics

### ISSUE THREAD

**Issue by Sabuto** [dgreif/ring#682]
# Bug Report

### Describe the Bug
 A clear and concise description of the problem...
I am trying to get started and it's throwing this error whenever i try and build

### To Reproduce
 - Create an `ApiClient`
 - Attempt to get locations and log them out
 - See error

### Expected behavior
 A clear and concise description of what you expected to happen.

### Screenshots/Logs
```node_modules/ring-client-api/lib/api/location.d.ts:47:30 - error TS2503: Cannot find namespace 'SocketIOClient'.

47     getConnection(): Promise<SocketIOClient.Socket>;
                                ~~~~~~~~~~~~~~


Found 1 error.```

### Additional context
<!-- ✍️edit: --> Add any other context about the problem here.

### Environment
<!-- ✍️edit: -->
 - OS: [e.g. Raspbian]
 - Node.js: [e.g. 12.14.1] <!-- run `node -v` to get current version -->
 - NPM: [e.g 6.13.4] <!-- run `npm -v` to get current version -->
 - ring-client-api: [e.g 6.0.3]

**Comment by dgreif** [comment]
Sorry, but there is a lot of information missing here.  This _looks_ like a typescript error, but without more info from there error and an understanding of your current setup, there isn't much I can do to help

**Comment by Sabuto** [comment]
hi sorry, this i expect it is because of typescript, 

this is my index.ts

```typescript
/* tslint:disable */
import {RingApi} from 'ring-client-api'

const ringApi = new RingApi({
    refreshToken: actual-token-here,
    debug: true
})

const mainFunction = async(ringApi: RingApi) => {
    const locations = await ringApi.getLocations()
    const location = locations[0]

    console.log(location.cameras)
}

mainFunction(ringApi)
```

my package.json
```
{
  "name": "RingApi",
  "version": "0.0.1",
  "description": "Testing the ring api, hopefully to integrate with hassio",
  "main": "index.js",
  "scripts": {
    "start": "tsc && node output/index.js"
  },
  "license": "MIT",
  "devDependencies": {
    "homebridge": "^1.3.4",
    "socket.io-client": "2.4.0",
    "typescript": "^4.3.4"
  },
  "dependencies": {
    "ring-client-api": "9.18.1"
  }
}
```
and my tsconfig
```typescript
{
    "compilerOptions": {
        "module": "commonjs",
        "esModuleInterop": true,
        "target": "es2015",
        "lib": [
            "es5",
            "ES2015",
            "es2017",
            "DOM"
        ],
        "moduleResolution": "node",
        "sourceMap": false,
        "outDir": "output"
    }
}
```

I am running on windows wsl 2 debian, Node: v14.16.0 npm : v14.16.0

**Comment by dgreif** [comment]
You can probably remove `homebridge` and `socket.io-client` from your `devDependencies`.  `socket.io-client` does not actually include the types (they are in `@types/socket.io-client`, which is likely why you are getting an error for that particular type.  I'd remove those, then delete your `node_modules` folder and run `npm i`.

**Comment by Sabuto** [comment]
i will try this, when i didnt have the homebridge in i was getting another type error but i will try this tomorrow and report back thanks :)

**Comment by Sabuto** [comment]
so with an updated package.json
```
{
  "name": "RingApi",
  "version": "0.0.1",
  "description": "Testing the ring api, hopefully to integrate with hassio",
  "main": "index.js",
  "scripts": {
    "start": "tsc && node output/index.js"
  },
  "license": "MIT",
  "devDependencies": {
    "typescript": "^4.3.4"
  },
  "dependencies": {
    "ring-client-api": "9.18.1"
  }
}
```

i get the following errors
```
node_modules/@homebridge/camera-utils/lib/return-audio-transcoder.d.ts:3:38 - error TS2307: Cannot find module 'homebridge' or its corresponding type declarations.

3 import { PrepareStreamRequest } from 'homebridge';
                                       ~~~~~~~~~~~~

node_modules/ring-client-api/lib/api/location.d.ts:47:30 - error TS2503: Cannot find namespace 'SocketIOClient'.

47     getConnection(): Promise<SocketIOClient.Socket>;
                                ~~~~~~~~~~~~~~


Found 2 errors.
```

**Comment by veverkap** [comment]
I'm getting the same errors :(

**Comment by veverkap** [comment]
Literally copied the package.json and ran:

```
➜  ring-downloader git:(master) ✗ rm -rf node_modules
➜  ring-downloader git:(master) ✗ yarn
yarn install v1.22.10
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
warning ring-client-api@9.18.1: The engine "homebridge" appears to be invalid.
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
✨  Done in 3.71s.
➜  ring-downloader git:(master) ✗ tsc  server.ts
node_modules/@homebridge/camera-utils/lib/return-audio-transcoder.d.ts:3:38 - error TS2307: Cannot find module 'homebridge' or its corresponding type declarations.

3 import { PrepareStreamRequest } from 'homebridge';
                                       ~~~~~~~~~~~~

node_modules/ring-client-api/lib/api/location.d.ts:47:30 - error TS2503: Cannot find namespace 'SocketIOClient'.

47     getConnection(): Promise<SocketIOClient.Socket>;
                                ~~~~~~~~~~~~~~


Found 2 errors.

➜  ring-downloader git:(master) ✗

**Comment by Sabuto** [comment]
i changed from ts to js and it worked for me

**Comment by jonathan-reisdorf** [comment]
As a workaround for the missing types, `npm install --save-dev @types/socket.io-client@1.4.36` should work.

**Comment by dgreif** [comment]
Hey all, sorry for all the trouble!  There were definitely a few type dependencies missing from the `ring-client-api` package.  I got those all fixed up in `9.18.2` if you want to give it another try.  I also created a sample project to show basic setup, and as a place that I can test this in the future for my own sanity 😄: https://github.com/dgreif/ring-client-example
